repository/git.go

```diff
@@ -407,12 +427,61 @@ func (r *Repository) commitWorktreeChanges(ctx context.Context, worktreePath, ex
 // AI slop below!
 // this is just to keep us moving fast because big git repos get hard to work with
 // and our demos like to download large dependencies.
-func (r *Repository) addNonBinaryFiles(ctx context.Context, worktreePath string) error {
+// getSubmodulePaths returns a slice of submodule paths relative to the worktree root
+func (r *Repository) getSubmodulePaths(ctx context.Context, worktreePath string) []string {
+	submodulePaths := []string{}
+
+	// Check if .gitmodules exists
+	gitmodulesPath := filepath.Join(worktreePath, ".gitmodules")
+	if _, err := os.Stat(gitmodulesPath); os.IsNotExist(err) {
+		return submodulePaths
+	}
+
+	// Use git submodule foreach to get submodule paths
+	output, err := RunGitCommand(ctx, worktreePath, "submodule", "foreach", "--recursive", "--quiet", "echo $sm_path")
+	if err != nil {
+		// If command fails, return empty slice - don't block regular operation
+		slog.Debug("Failed to get submodule paths", "error", err)
+		return submodulePaths
+	}
+
+	for line := range strings.SplitSeq(strings.TrimSpace(output), "\n") {
```

Comment:
🔧 **Maintainability (Medium):** **Inconsistent submodule detection logic**: Two different implementations exist for detecting submodule paths - one in `environment/filesystem.go:detectSubmodulePaths()` and another in `repository/git.go:getSubmodulePaths()`. They use different approaches:

**repository/git.go** (line 448): Uses `strings.SplitSeq()` with range iteration
**environment/filesystem.go** (line 227): Uses `strings.Split()` with index iteration

**Risk**: These subtle differences could lead to inconsistent behavior between host and container submodule detection.

```suggestion
	for _, line := range strings.Split(strings.TrimSpace(output), "\n") {
```

Reply:
dead code, deleted